front/src/modules/timesStops/hooks/useOutputTableData.ts

```diff
@@ -79,18 +79,20 @@ function useOutputTableData(
         );
         const pathStepKey = `${sugOpPoint.uic}-${sugOpPoint.ch}`;
 
-        if (pathStepKey in pathStepsByUic && nextOpPoint) {
+        if (pathStepKey in pathStepsByUic) {
           const pathStepId = pathStepsByUic[pathStepKey].id || '';
           const schedule = scheduleByAt[pathStepId];
           const scheduleData = computeScheduleData(schedule, selectedTrainSchedule.start_time);
           const formattedScheduleData = formatScheduleData(scheduleData);
-          const marginsData = computeMargins(
-            simulatedTrain,
-            opPoint,
-            nextOpPoint,
-            selectedTrainSchedule,
-            pathStepId
-          );
+          const marginsData = nextOpPoint
```

Comment:
Can you do that inside the computeMargins function and make a new unit test for this new case ?

Reply:
Hm, does it really make sense? `computeMargins()` is non-sensical without a `nextOpPoint` (would return immediately with an empty object), or is it?

Reply:
I think it would make more sense to let the function handle that case when there is no `nextOpPoint` rather than having a ternary at the call site here

Reply:
Can you expand why? To me, it's the contrary since the function can't do anything without a `nextOpPoint` :P

Reply:
well :P
- mostly because it’s impossible to test here in this hook, and it’s really easy to test the function directly.
- In the way I view function design, as a general principal, it would be the function’s responsibility to handle the case where their input is undefined: It looks a lot messier to have this 
`nextOpPoint ? myFunction(nextOpPoint) : null;` rather than hide it inside `myFunction(nextOpPoint)`

Reply:
I agree more with @emersion  approach. The `null` check is really trivial. It makes more sense to test the whole hook, but we can discuss that later.